src/EnergyPlus/ZoneTempPredictorCorrector.cc

```diff
@@ -3985,6 +4079,109 @@ namespace ZoneTempPredictorCorrector {
+
+						// Calculate MCPI used for hybrid modeling included in TempIndCoef and TempDepCoef
+						Real64 a = Zone( ZoneNum ).OutDryBulbTemp;
+						Real64 b = SumIntGain + SumHATsurf - SumHATref + MCPTV( ZoneNum ) + MCPTM( ZoneNum ) + MCPTE( ZoneNum ) + MCPTC( ZoneNum ) + MDotCPOA( ZoneNum ) * Zone( ZoneNum ).OutDryBulbTemp;
+						Real64 c = SumHA + MCPV( ZoneNum ) + MCPM( ZoneNum ) + MCPE( ZoneNum ) + MCPC( ZoneNum ) + MDotCPOA( ZoneNum );
+						Real64 d = AirCap;
+						//Use3rdOrder
+						Real64 AA = b + d * ( 3.0 * PreviousMeasuredZT1( ZoneNum ) - ( 3.0 / 2.0 ) * PreviousMeasuredZT2( ZoneNum ) + ( 1.0 / 3.0 ) * PreviousMeasuredZT3( ZoneNum ) );
+						Real64 BB = ( 11.0 / 6.0 ) * d + c;
+						Real64 InfilVolumeOADensityHM;
+						Real64 AirDensity;
+						Real64 CpAir;
+						Real64 MCPIHM;
+						static std::string const RoutineNameInfiltration( "CalcAirFlowSimple:Infiltration" );
+
+     					CpAir = PsyCpAirFnWTdb( OutHumRat, Zone( ZoneNum ).OutDryBulbTemp );
+						AirDensity = PsyRhoAirFnPbTdbW( OutBaroPress, Zone( ZoneNum ).OutDryBulbTemp, OutHumRat, RoutineNameInfiltration );
+						MCPIHM = ( AA - Zone( ZoneNum ).ZoneMeasuredTemperature * BB ) / ( Zone( ZoneNum ).ZoneMeasuredTemperature - a ); // MCPIHM calculation using Use3rdOrder method 
+						InfilVolumeOADensityHM = ( MCPIHM / CpAir / AirDensity ) * TimeStepZone * SecInHour;
+						
+						if ( abs( Zone( ZoneNum ).ZoneMeasuredTemperature - Zone( ZoneNum ).OutDryBulbTemp ) > 5.0 && abs( ZT( ZoneNum ) - PreviousMeasuredZT1( ZoneNum ) ) < 0.1 ) {// Filter
+							InfilOAACHHM = InfilVolumeOADensityHM / ( TimeStepZone * Zone( ZoneNum ).Volume );
+							InfilOAACHHM = max( 0.0, min( 10.0, InfilOAACHHM ) );  // ACH max 10, min 0
+						} else {
+							InfilOAACHHM = 0.0;
+						}
+
+						Zone(ZoneNum).InfilOAAirChangeRateHM = InfilOAACHHM;
+						Zone(ZoneNum).MCPIHM = InfilOAACHHM * CpAir * AirDensity / SecInHour * Zone( ZoneNum ).Volume;
+
+					} // Hybrid model infiltration calcualtion end
+
+					// Hybrid modeling internal thermal mass calcualtion start
+					if ( HybridModelZone( ZoneNum).InternalThermalMassCalc && SumSysMCpT == 0 && ZT( ZoneNum ) != PreviousMeasuredZT1( ZoneNum ) && UseZoneTimeStepHistory ){ // HM calculation only when SumSysMCpT =0, TimeStepZone (not @ TimeStepSys)
+
+						// Calculate air capacity using UseAnalyticalSolution
+						if ( TempDepCoef == 0.0 ) {
+							AirCapHM = TempIndCoef / ( ZT( ZoneNum ) - PreviousMeasuredZT1( ZoneNum ) ); // Inverse equation
+						}
+						else {
+							Real64 AirCapHM_temp = ( TempIndCoef - TempDepCoef * PreviousMeasuredZT1( ZoneNum ) ) / ( TempIndCoef - TempDepCoef * ZT( ZoneNum ) );
+							if ( (AirCapHM_temp > 0 ) && ( AirCapHM_temp != 1 ) ){ // Avoide IND
+								AirCapHM = TempDepCoef / std::log( AirCapHM_temp ); // Inverse equation
+							}
+							else {
+								AirCapHM = TempIndCoef / ( ZT( ZoneNum ) - PreviousMeasuredZT1( ZoneNum ) );
```

Comment:
I addressed all comments. No errors on my local machine. I synced to the github for another round of CI tests. Fingers crossed.